test/e2e/catalog_e2e_test.go

```diff
@@ -1028,6 +1032,165 @@ var _ = Describe("Catalog represents a store of bundles which OLM can use to ins
+			if err != nil {
+				return false, err
+			}
+			// if the conditions array has the entry we know things got updated
+			condition := meta.FindStatusCondition(source.Status.Conditions, catalogtempate.StatusTypeTemplatesHaveResolved)
+			if condition != nil {
+				return true, nil
+			}
+
+			return false, nil
+		}, 5*time.Minute, 1*time.Second).Should(BeTrue())
+
+		// source should be the latest we got from the eventually block
+		Expect(source.Status.Conditions).ToNot(BeNil())
+
+		/*
+			///////
+			// FIXME: There have been concerns about security of allowing selection of arbitrary
+			// GVK during code reviews. For now we're disabling setting up a gvk template. This
+			// code needs to be re-enabled once we've come up with a valid approach
+			// the first time a GVK template is encountered, watches are setup for those resources so the
+			// templates can't be resolved immediately. We should have status conditions and they should indicate
+			// that resolution has not happened yet
+			templatesResolvedCondition := meta.FindStatusCondition(source.Status.Conditions, catalogtempate.StatusTypeTemplatesHaveResolved)
+			if Expect(templatesResolvedCondition).ToNot(BeNil()) {
+				Expect(templatesResolvedCondition.Reason).To(BeIdenticalTo(catalogtempate.ReasonUnableToResolve))
+				Expect(templatesResolvedCondition.Status).To(BeIdenticalTo(metav1.ConditionFalse))
+			}
+			resolvedImageCondition := meta.FindStatusCondition(source.Status.Conditions, catalogtempate.StatusTypeResolvedImage)
+			if Expect(resolvedImageCondition).ToNot(BeNil()) {
+				Expect(resolvedImageCondition.Reason).To(BeIdenticalTo(catalogtempate.ReasonUnableToResolve))
+				Expect(resolvedImageCondition.Status).To(BeIdenticalTo(metav1.ConditionFalse))
+			}
+
+			// update catalog source with annotation (to kick resync)
+			source, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Get(context.TODO(), source.GetName(), metav1.GetOptions{})
+			Expect(err).ShouldNot(HaveOccurred(), "error getting catalog source pod")
+			source.Annotations["testKey"] = genName("newValue")
+			_, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Update(context.TODO(), source, metav1.UpdateOptions{})
+			Expect(err).ShouldNot(HaveOccurred(), "error updating catalog source pod with test annotation")
+
+			// wait for change in status condition
+			Eventually(func() (bool, error) {
+				source, err = crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Get(context.TODO(), sourceName, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				// condition should flip to true indicating that the catalog image has been updated
+				if meta.IsStatusConditionTrue(source.Status.Conditions, catalogtempate.StatusTypeTemplatesHaveResolved) {
+					return true, nil
+				}
+
+				return false, nil
+			}, 5*time.Minute, 1*time.Second).Should(BeTrue())
+			///////
+		*/
```

Comment:
Can we remove this in a cleanup commit? That way, we can retain the code in git without impacting readability/understandability.

Reply:
fixed